### PR TITLE
Disable File related functionality

### PR DESCRIPTION
**Pull request by szysad**
## Before submitting checklist

- [x] Did you **update the CHANGELOG**? (not for test updates, internal changes/refactors or CI/CD setup)
- [ ] Did you **ask the docs owner** to review all the user-facing changes?

### CODE REVIEW

**Comment by szysad** [comment]
[E2E](https://github.com/neptune-ai/neptune-client/actions/runs/8602827915)

**Comment by szysad** [comment]
[E2E](https://github.com/neptune-ai/neptune-client/actions/runs/8613898832)

File: tests/e2e/base.py [hunk 1]
```diff
@@ -37,6 +46,51 @@
 ]
 
 
+class Parameters:
+    param_d: Dict[str, Any]
+
+    def __init__(self, params: List[ParameterSet]) -> "Parameters":
```
Comment (AleksanderWWW) [inline tests/e2e/base.py:52]: The correct type hint for `__init__` return is `None` - this method never returns anything

File: tests/e2e/base.py [hunk 2]
```diff
@@ -37,6 +46,51 @@
 ]
 
 
+class Parameters:
+    param_d: Dict[str, Any]
```
Comment (AleksanderWWW) [inline tests/e2e/base.py:50]: Is this needed? I don't see any usage of this as a class attribute - only instance. Maybe we could add a type hint in the constructor?

File: tests/e2e/base.py [hunk 3]
```diff
@@ -37,6 +46,51 @@
...
+        )
+
+    def run(self, *args: str) -> "Parameters":
+        return self._modify(*args, func=lambda x: pytest.param(x))
+
+    def eval(self) -> List[ParameterSet]:
+        return list(self.param_d.values())
+
+
+class ParametersFactory:
```
Comment (AleksanderWWW) [inline tests/e2e/base.py:77]: I'm wondering if it makes sense to have it as a class vs. to just have to functions - we don't have any state here, right? Just using it as a namespace for the methods

File: tests/e2e/base.py [hunk 4]
```diff
@@ -37,6 +46,51 @@
...
+
+    @staticmethod
+    def custom(params: List[Union[str, ParameterSet]]) -> Parameters:
+        normalized = []
+        for p in params:
+            if isinstance(p, str):
+                normalized.append(pytest.param(p))
+            else:
+                normalized.append(p)
```
Comment (AleksanderWWW) [inline tests/e2e/base.py:86]:
```suggestion
        normalized = [pytest.param(p) if isinstance(p, str) else p for p in params]
```

How about this?

**Review by AleksanderWWW — COMMENTED** [review]
Some minor comments, otherwise looks good :rocket:

File: tests/e2e/standard/test_series.py [hunk 5]
```diff
@@ -37,7 +39,11 @@
     tmp_context,
 )
 
-BASIC_SERIES_TYPES = ["strings", "floats", "files"]
+BASIC_SERIES_TYPES = (
+    ParametersFactory.custom(["strings", "floats", "files"])
+    .xfail("files", reason="File funcitonality disabled", raises=NeptuneUnsupportedFunctionalityException)
```
Comment (AleksanderWWW) [inline tests/e2e/standard/test_series.py:44]: functionality* (typo) :wink:

File: tests/e2e/base.py (hunk 2 above)
Comment (szysad) [reply tests/e2e/base.py:50]: you are right. I always thought of this as type hint for class instance attribute. I guess it makes sense this is type hint but for whole class scope

**Review by AleksanderWWW — APPROVED** [review]

**Pull request merged**